### ISSUE TITLE
error: 'memset' is not a member of 'std';

### ISSUE THREAD

**Issue by jq2333** [ros/meta-ros#1130]
**Describe the bug**
* yocto branch: kirkstone 4.0.17
* branch:kirkstone
* rev:685215f9837d47089d2eaaa296c1dac574f44ddf
* ros2-foxy
* meta-ros/meta-ros2-foxy/generated-recipes/rmw-cyclonedds/rmw-cyclonedds-cpp_0.7.7-1
* gcc12.3.0

**To Reproduce**
use gcc12.3.0 to compile.

**Expected behavior**
build pass

**Solution**
```
diff --git a/src/Serialization.cpp b/src/Serialization.cpp
index 8263c58..2ab6931 100644
--- a/src/Serialization.cpp
+++ b/src/Serialization.cpp
@@ -20,6 +20,8 @@

 #include "Serialization.hpp"

+#include <cstring> //std::memcpy
+
 #include <array>
 #include <limits>
 #include <memory>
```

https://github.com/ros2/rmw_cyclonedds/blob/humble/rmw_cyclonedds_cpp/src/Serialization.cpp

**Comment by robwoolley** [comment]
Thanks for reporting this.

The ROS 2 Foxy release went end-of-life on June 20, 2023.  Unfortunately, it is no longer supported.

It appears that this was fixed in Humble: https://github.com/ros2/rmw_cyclonedds/commit/6bc618534e6230f0c6db102c8a7b0706e902c763

If there is something that is keeping you on Foxy, I'd be interested in understanding why you can't migrate forward.

**Comment by jq2333** [comment]
> Thanks for reporting this.
> 
> The ROS 2 Foxy release went end-of-life on June 20, 2023. Unfortunately, it is no longer supported.
> 
> It appears that this was fixed in Humble: [ros2/rmw_cyclonedds@6bc6185](https://github.com/ros2/rmw_cyclonedds/commit/6bc618534e6230f0c6db102c8a7b0706e902c763)
> 
> If there is something that is keeping you on Foxy, I'd be interested in understanding why you can't migrate forward.

Thank you!
I'll migrate to humble as soon as possible！！
